gobblin-runtime/src/main/java/org/apache/gobblin/runtime/task/BaseAbstractTask.java

```diff
@@ -91,4 +94,13 @@ public String getProgress() {
   public boolean isSpeculativeExecutionSafe() {
     return false;
   }
+
+  /**
+   * Similar to org.apache.gobblin.runtime.Task#failTask(java.lang.Throwable), we need to propagate exception thrown
+   * in workunit level by setting {@link ConfigurationKeys#TASK_FAILURE_EXCEPTION_KEY}
+   */
+  protected void failTask(Throwable t, TaskContext taskContext) {
+    taskContext.getTaskState().setWorkingState(WorkUnitState.WorkingState.FAILED);
```

Comment:
Should we save taskContext in BaseAbstractTask ? In this way, signature of this method can be consistent with that in Task.java